File: CXOCStudy/CXOCStudy/design-mode/Prototype/CXPrototype.m
//
//  CXPrototype.m
//  CXOCStudy
//
//  Created by xiaoma on 2017/11/15.
//  Copyright © 2017年 CX. All rights reserved.
//

#import "CXPrototype.h"

@implementation CXPrototype

- (instancetype)init
{
    self = [super init];
    if (self) {
        self.content = @"Dog";
    }
    return self;
}

- (id)copyWithZone:(NSZone *)zone {
    CXPrototype *copyLayer = [[[self class] allocWithZone:zone] init];
    copyLayer.background = self.background;
    copyLayer.content = self.content;
    return copyLayer;
}

@end

File: CXSwiftStudy/CXSwiftStudy/design-mode/CXPrototype.swift
//
//  CXPrototype.swift
//  CXSwiftStudy
//
//  Created by xiaoma on 2017/11/14.
//  Copyright © 2017年 CX. All rights reserved.
//

import UIKit

class CXSimpleLayer: NSObject, NSCopying {
    
    
    var background = [0, 0, 0, 0]
    var content = "blank"
    
    init(background: Array<Any>, content: String) {
        self.background = background as! [Int];
        self.content = content;
    }
    
    func copy(with zone: NSZone? = nil) -> Any {
        return CXSimpleLayer(background: background, content: content)
    }
}



